frontend/tests/features/wheresMyGene/tissueAutoExpand.test.ts

```diff
@@ -0,0 +1,212 @@
+  });
+  /**
+   * Filter 3 tissues, filter top 3 cell types
+   * (B cell, B-1a B cell, and B-1b B cell). Expect both tissues expanded.
+   * Remove “B cell” from filter <-- this removes tissue filter, since only
+   * “Lung” tissue has B-1a B cell and B-1b B cell
+   */
+  test("Filter cell type auto expansion - remove cell type filter", async ({
+    page,
+  }) => {
+    const cells = CELL_TYPE_FILTERS.slice(1, 3);
+    await loadPageAndTissues(page);
+    await filterTissues(page);
+    await filterCellType(page);
+    await checkTissues(page);
+    await removeCellFilter(page);
+    await checkTissues(page, ["lung"], []);
+    await checkCellTypes(page, cells);
+  });
+});
+/**
+ * *******************************************
+ * Helper Functions
+ * *******************************************
+ */
+
+/**
+ * loadPageAndTissues
+ * Load the WMG page and wait for the tissue nodes to be visible
+ */
+async function loadPageAndTissues(page: Page) {
+  await goToWMG(page);
+  await expect(page.getByTestId(TISSUE_NODE_TEST_ID)).not.toHaveCount(0);
+}
+
+/**
+ * clickIntoFilter
+ * Click into the filter and wait for the tooltip to be visible
+ */
+async function clickIntoFilter(page: Page, filterName: string) {
+  await page
+    .getByTestId(filterName)
+    .getByRole("button", { name: TISSUE_FILTER_LABEL, exact: true })
+    .click();
+  await page.getByRole("tooltip").waitFor();
+}
+
+/**
+ * filterCellType
+ * Filter cell type 'B cell' and exit filter mode
+ */
+async function filterCellType(page: Page, count = 3) {
+  await page
+    .getByTestId(CELL_TYPE_FILTER_TEST_ID)
+    .getByRole("combobox")
+    .click();
+  for (let i = 0; i < count; i++) {
+    await page
+      .getByRole("option", { name: CELL_TYPE_FILTERS[i], exact: true })
+      .click();
+  }
+  await page.keyboard.press("Escape");
+}
+
+/**
+ * filterTissues
+ * Filter the tissuesfrom the left panel
+ */
+async function filterTissues(
+  page: Page,
+  filteredTissues: string[] = FILTERED_TISSUES
+) {
+  await clickIntoFilter(page, TISSUE_FILTER_TEST_ID);
+  for (const tissue of filteredTissues) {
+    await page.getByRole("option", { name: tissue, exact: true }).click();
+  }
+  await page.keyboard.press("Escape");
+}
+
+/**
+ * checkTissues
+ * Check that only filtered tissues are visible, and expanded tissues are expanded
+ */
+async function checkTissues(
+  page: Page,
+  filteredTissues: string[] = FILTERED_TISSUES,
+  expandedTissues = filteredTissues
+) {
+  if (filteredTissues.length !== 0) {
+    await expect(page.getByTestId("tissue-name")).toHaveCount(
+      filteredTissues.length
+    );
+  }
+
+  let i = 0;
+  for (const tissue of expandedTissues) {
+    await expect(page.getByTestId(`cell-type-labels-${tissue}`)).toBeVisible();
+    const height = await page
+      .getByTestId(`cell-type-labels-${tissue}`)
+      .getAttribute("height");
+    toInteger(height) > 20 && i++;
+  }
+  await expect(i).toEqual(expandedTissues.length);
+}
```

Comment:
discussion:

What's your thought on using `index` from `expandedTissues`, so we don't need to track `let i`?

Like so:
```ts
for (const [index, tissue] of Object.entries(expandedTissues)) {
  ...
}
```

Reply:
How would you make the assertion outside of the loop? the usage of `i` here is a counter